openedx/core/djangoapps/catalog/utils.py

```diff
@@ -135,3 +135,36 @@ def get_programs_with_type(types=None):
             programs_with_type.append(program_with_type)
 
     return programs_with_type
+
+
+def get_catalog_course_runs():
+    """
+    Retrieve all the course runs from the catalog service.
+
+    Returns:
+        list of dict with each record representing a course run.
+    """
+    catalog_integration = CatalogIntegration.current()
+    course_runs = []
+    if catalog_integration.enabled:
+        try:
+            user = User.objects.get(username=catalog_integration.service_username)
+        except User.DoesNotExist:
+            return course_runs
+
+        api = create_catalog_api_client(user, catalog_integration)
+
+        querystring = {
+            'page_size': catalog_integration.page_size,
```

Comment:
I think let's go with 100 page size and monitor it after roll out to see if we need change the size.